Const to non-const conversion error

Const to non-const conversion error

## Issue Details

I tried to compile the example for incremental searching (file: Spatial_searching/distance_browsing.cpp) with a small modification (line 30):

`std::cout <<   (*it).first << "  at squared distance = " << (*it).second << std::endl;`

to

`std::cout <<   it->first << "  at squared distance = " << it->second << std::endl;`

Then, a compiler error occured:

`C:\lib\CGAL\include\CGAL\iterator.h(618,1): error C2440: 'return': cannot convert from 'const std::pair<CGAL::Point_2<Kernel_>,double> *' to 'std::pair<CGAL::Point_2<Kernel_>,double> *'`

It seams to be a bug on the member access operator (->) of the class iterator.

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Windows 10 18363.836, x64 bits
* Compiler: MSVC 19.25.28614 for x64
* Release or debug mode: Release
* Specific flags used (if any): CGAL_HEADER_ONLY
* CGAL version: 4.14
* Boost version: 1.70
* Other libraries versions if used (Eigen, TBB, etc.):

--- a/Spatial_searching/examples/Spatial_searching/distance_browsing.cpp
+++ b/Spatial_searching/examples/Spatial_searching/distance_browsing.cpp
@@ -35,7 +35,7 @@ int main() {
 
   std::cout <<  "The first 5 nearest neighbours with positive x-coord are: " << std::endl;
   for (int j=0; (j < 5)&&(it!=end); ++j,++it)
-    std::cout <<   (*it).first << "  at squared distance = " << (*it).second << std::endl;
+    std::cout <<   (*it).first << "  at squared distance = " << it->second << std::endl;
 
   return 0;
 }

--- a/Spatial_searching/include/CGAL/Incremental_neighbor_search.h
+++ b/Spatial_searching/include/CGAL/Incremental_neighbor_search.h
@@ -147,7 +147,7 @@ namespace CGAL {
 
       typedef std::input_iterator_tag iterator_category;
       typedef Point_with_transformed_distance       value_type;
-      typedef Point_with_transformed_distance*      pointer;
+      typedef const Point_with_transformed_distance*      pointer;
       typedef const Point_with_transformed_distance&      reference;
       typedef std::size_t               size_type;
       typedef std::ptrdiff_t            difference_type;

--- a/Spatial_searching/include/CGAL/Orthogonal_incremental_neighbor_search.h
+++ b/Spatial_searching/include/CGAL/Orthogonal_incremental_neighbor_search.h
@@ -480,7 +480,7 @@ namespace CGAL {
 
       typedef std::input_iterator_tag iterator_category;
       typedef Point_with_transformed_distance       value_type;
-      typedef Point_with_transformed_distance*      pointer;
+      typedef const Point_with_transformed_distance*      pointer;
       typedef const Point_with_transformed_distance&      reference;
       typedef std::size_t               size_type;
       typedef std::ptrdiff_t            difference_type;

